javascript/ql/src/semmle/javascript/frameworks/NodeJSLib.qll

```diff
@@ -478,6 +478,28 @@ module NodeJSLib {
               DataFlow::moduleImport("util-promisifyall")
             ].getACall()
         )
+        or
+        // const fs = require('fs');
+        // module.exports = methods.reduce((obj, method) => {
+        //  obj[method] = fs[method];
+        //  return obj;
+        // }, {});
+        t.continue() = t2 and
+        exists(
```

Comment:
This step through `reduce` seems general enough that it belongs in its own reusable predicate, could the  `fsModule` and `maybePromisified` predicates be factored out?

Reply:
We could refactor the second half of `fsModule` into a `maybePromisefyObject`.   
But I think that is something we can look into when we find another model that needs the same treatment.